graphql/authorization/auth.go

```diff
@@ -39,33 +43,56 @@ var (
 )
 
 type AuthMeta struct {
-	PublicKey string
-	Header    string
-	Namespace string
+	HMACPublicKey string
+	RSAPublicKey  *rsa.PublicKey
+	Header        string
+	Namespace     string
+	Algo          string
 }
 
-func (m *AuthMeta) Parse(schema string) {
+func (m *AuthMeta) Parse(schema string) error {
 	poundIdx := strings.LastIndex(schema, "#")
 	if poundIdx == -1 {
-		return
+		return nil
 	}
 	lastComment := schema[poundIdx:]
-	length := strings.Index(lastComment, "\n")
-	if length > -1 {
-		lastComment = lastComment[:length]
+	if !strings.HasPrefix(lastComment, "# Authorization") {
+		return nil
+	}
+
+	authMetaRegex, err :=
+		regexp.Compile(`^#[\s]([^\s]+)[\s]+([^\s]+)[\s]+([^\s]+)[\s]+([^\s]+)[\s]+"([^\"]+)"`)
+	if err != nil {
+		return errors.Errorf("error while parsing jwt authorization info: %v", err)
+	}
+	idx := authMetaRegex.FindAllStringSubmatchIndex(lastComment, -1)
+	if len(idx) != 1 || len(idx[0]) != 12 ||
+		!strings.HasPrefix(lastComment, lastComment[idx[0][0]:idx[0][1]]) {
+		return errors.Errorf("error while parsing jwt authorization info")
 	}
 
-	authMeta := strings.Split(lastComment, " ")
-	if len(authMeta) != 5 || authMeta[1] != "Authorization" {
-		return
+	m.Header = lastComment[idx[0][4]:idx[0][5]]
+	m.Namespace = lastComment[idx[0][6]:idx[0][7]]
+	m.Algo = lastComment[idx[0][8]:idx[0][9]]
+
+	key := lastComment[idx[0][10]:idx[0][11]]
+	if m.Algo == "HS256" {
+		m.HMACPublicKey = key
+		return nil
 	}
-	m.Header = authMeta[2]
-	m.PublicKey = authMeta[3]
-	m.Namespace = authMeta[4]
+	if m.Algo != "RS256" {
```

Comment:
Did you mean RSA? If yes, please use RSA256
Same for H256, better to use HMAC256.